src/aiomealie/models.py

```diff
@@ -125,21 +128,35 @@ class ShoppingListsResponse(DataClassORJSONMixin):
 
 
 @dataclass
-class ShoppingItem(DataClassORJSONMixin):
+class ShoppingItem(DataClassORJSONMixin, DataClassDictMixin):
     """ShoppingItem model."""
 
-    item_id: str = field(metadata=field_options(alias="id"))
-    list_id: str = field(metadata=field_options(alias="shoppingListId"))
-    note: str
-    display: str
-    checked: bool
-    position: int
-    is_food: bool = field(metadata=field_options(alias="isFood"))
-    disable_amount: bool = field(metadata=field_options(alias="disableAmount"))
-    quantity: float
-    label_id: str = field(metadata=field_options(alias="labelId"))
-    food_id: str = field(metadata=field_options(alias="foodId"))
-    unit_id: str = field(metadata=field_options(alias="unitId"))
+    item_id: Optional[str] = field(default=None, metadata=field_options(alias="id"))
+    list_id: Optional[str] = field(
+        default=None, metadata=field_options(alias="shoppingListId")
+    )
+    note: Optional[str] = field(default=None)
+    display: Optional[str] = field(default=None)
+    checked: Optional[bool] = field(default=None)
+    position: Optional[int] = field(default=None)
+    is_food: Optional[bool] = field(
+        default=None, metadata=field_options(alias="isFood")
+    )
+    disable_amount: Optional[bool] = field(
+        default=None, metadata=field_options(alias="disableAmount")
+    )
+    quantity: Optional[float] = field(default=None)
+    label_id: Optional[str] = field(
+        default=None, metadata=field_options(alias="labelId")
+    )
+    food_id: Optional[str] = field(default=None, metadata=field_options(alias="foodId"))
+    unit_id: Optional[str] = field(default=None, metadata=field_options(alias="unitId"))
+
+    class Config(BaseConfig):  # pylint: disable=too-few-public-methods
+        """Mashumaro Config."""
+
+        serialize_by_alias = True
+        code_generation_options = ["TO_DICT_ADD_OMIT_NONE_FLAG"]
```

Comment:
Hmm, I think it would be better if we split the objects and have a `CreateShoppingItem` object. This would save us a lot of `assert item.something` in the code because everything is now nullable.

WDYT?

Reply:
I did start off with two but thought it was duplicating it.  The `CreateShoppingItem` would still need all the optional/default none.
Where are you thinking the asserts will be need, on the get list?
Also `CreateShoppingItem` would also be used for Updates, so any suggestion on a better name?

Reply:
`MutateShoppingItem`

It's more that if we want to use it in Home assistant, that if we want to use the value where they need a `str`, the value we have currently is typed `str | None`, so we have to `assert something is not None` before mypy sees it as `str`.

Also, please use the union type over the Optional `str | None` > `Optional[str]`